lib/CL/devices/pthread/pthread_scheduler.c

```diff
@@ -361,77 +347,75 @@ work_group_scheduler (kernel_run_command *k,
 
 static int
 work_group_scheduler (kernel_run_command *k,
-                     struct pool_thread_data *thread_data)
+                      struct pool_thread_data *thread_data)
 {
   cl_kernel kernel = k->kernel;
   cl_program program = kernel->program;
   pocl_kernel_metadata_t *meta = k->kernel->meta;
 
 #pragma omp parallel
-{
-  void *arguments[meta->num_args + meta->num_locals + 1];
-  void *arguments2[meta->num_args + meta->num_locals + 1];
-  struct pocl_context pc;
-  void *local_mem = malloc(scheduler.local_mem_size);
+  {
+    void *arguments[meta->num_args + meta->num_locals + 1];
+    void *arguments2[meta->num_args + meta->num_locals + 1];
+    struct pocl_context pc;
+    void *local_mem = malloc (scheduler.local_mem_size);
 
-  setup_kernel_arg_array_with_locals (
-      (void **)&arguments, (void **)&arguments2, k, local_mem,
-      scheduler.local_mem_size);
-  memcpy (&pc, &k->pc, sizeof (struct pocl_context));
+    setup_kernel_arg_array_with_locals ((void **)&arguments,
+                                        (void **)&arguments2, k, local_mem,
+                                        scheduler.local_mem_size);
+    memcpy (&pc, &k->pc, sizeof (struct pocl_context));
 
 #ifndef ENABLE_PRINTF_IMMEDIATE_FLUSH
-  assert (pc.printf_buffer_capacity > 0);
-  pc.printf_buffer = malloc (pc.printf_buffer_capacity);
-  assert (pc.printf_buffer != NULL);
-  uint32_t position = 0;
-  pc.printf_buffer_position = &position;
+    assert (pc.printf_buffer_capacity > 0);
+    pc.printf_buffer = malloc (pc.printf_buffer_capacity);
+    assert (pc.printf_buffer != NULL);
+    uint32_t position = 0;
+    pc.printf_buffer_position = &position;
 #else
-  pc.printf_buffer = NULL;
-  pc.printf_buffer_position = NULL;
+    pc.printf_buffer = NULL;
+    pc.printf_buffer_position = NULL;
 #endif
 
-  unsigned rm = pocl_save_rm ();
-  pocl_set_default_rm ();
-  unsigned ftz = pocl_save_ftz ();
-  pocl_set_ftz (program->flush_denorms);
+    unsigned rm = pocl_save_rm ();
+    pocl_set_default_rm ();
+    unsigned ftz = pocl_save_ftz ();
+    pocl_set_ftz (program->flush_denorms);
 
-  size_t x, y, z;
-  /* runtime = set scheduling according to environment variable OMP_SCHEDULE */
+    size_t x, y, z;
+    /* runtime = set scheduling according to environment variable OMP_SCHEDULE
+     */
 #pragma omp for collapse(3) schedule(runtime)
```

Comment:
adding parallel there would make the parallel construct recursive. i'm not sure that's a good idea, if it even works, and i can't remove the former parallel pragma, because we need per-thread allocated memory before the for loop.

according to the link you sent, "pragma omp parallel for" is the same as "pragma omp parallel" followed by "pragma omp for".